Graphite Plugin: Remove angular dependencies for graphite annotations

bohandley opened the pull request:
Fixes https://github.com/grafana/grafana/issues/50458

**What this PR does / why we need it**: This PR removes angular dependencies for graphite, creates annotations with a react component, and migrates the legacy annotations.

**Special notes for your reviewer**:
To test this PR, do the following:

1. Check out main.
2. Create a dashboard with a graphite datasource
3. Create an annotation with a tag or a query.
4. Save the Dashboard.
5. Check that the annotation is displayed on the dashboard.
6. Checkout this branch.
7. Check that the previous annotation is still displayed.
8. Create a new annotation and check that it is displayed.

bohandley committed:
handle new creating annotations

yesoreyeram commented on public/app/plugins/datasource/graphite/components/AnnotationsEditor.tsx:
```diff
@@ -0,0 +1,43 @@
+          placeholder="Example: statsd.application.counters.*.count"
+        />
+      </div>
+      <div className="gf-form">
```
In angular editor, there was `<h5 class="section-heading">Or</h5>` which is missing here.

bohandley replied:
Thank you for noticing this! I added the 'Or'.

yesoreyeram commented on public/app/plugins/datasource/graphite/datasource.ts:
```diff
@@ -326,13 +360,41 @@ export class GraphiteDatasource
-    if (options.annotation.target) {
-      const target = this.templateSrv.replace(options.annotation.target, {}, 'glob');
+  annotationEvents(range: any, target: GraphiteQuery) {
+    if (target.eventsQuery?.tags) {
```
Looks like a regression. The original behaviour if both target and tags set, target with take precedence over tags.

bohandley replied:
This makes sense, thank you. If someone had added an annotation with both a target and tags, my old code would have broken their annotation because it would have read the tags instead, causing the regression. I switched it so that it checks for target first to fix the regression.

yesoreyeram commented on public/app/plugins/datasource/graphite/module.ts:
```diff
@@ -5,12 +5,7 @@ import { MetricTankMetaInspector } from './components/MetricTankMetaInspector';
 import { GraphiteDatasource } from './datasource';
 
-class AnnotationsQueryCtrl {
-  static templateUrl = 'partials/annotations.editor.html';
```
make sure to remove the corresponding html file

bohandley replied:
Removed :) [1eab44e](https://github.com/grafana/grafana/pull/50453/commits/1eab44e6cca3a17a40ac74eff803fd3eb78ecb77)

yesoreyeram commented on public/app/plugins/datasource/graphite/types.ts:
```diff
@@ -6,8 +6,15 @@ import { GraphiteDatasource } from './datasource';
+export type GraphiteEventsQuery = {
+  fromAnnotations?: boolean;
+  target?: string;
+  tags?: string;
```
can we have tags typed as string[]?

bohandley replied:
I think so, I'll make this happen.

yesoreyeram commented on public/app/plugins/datasource/graphite/components/AnnotationsEditor.tsx:
```diff
@@ -0,0 +1,43 @@
+      </div>
+      <div className="gf-form">
+        <InlineFormLabel width={12}>Graphite events tags</InlineFormLabel>
+        <Input
```
Once if the tags type changed to string[], we can replace Input with TagsInput

bohandley replied:
I am looking into this, what you mention is similar to the way it is done in the Datadog plugin. I think the events() function in the datasource handles multiple tags. I can convert the input into an array and then handle it in the events function so we can, as you mentioned, replace Input with TagsInput. Thank you!

yesoreyeram commented on public/app/plugins/datasource/graphite/migrations.ts:
```diff
@@ -0,0 +1,33 @@
+      target: json.target,
+      tags: json.tags,
+    },
+  } as GraphiteQuery;
```
Instead of using `as GraphiteQuery`, can we return `"textEditor": true` if in case if the annotation is based on target string/query.

bohandley replied:
Yes, I think this makes sense. We can not use ```as GraphiteQuery``` and add "textEditor": true if the annotation is based on a target string query. The GraphiteQuery was originally one field, 'target', so maybe we can make a new type for the annotations?

bohandley committed:
add h5 'or' tag to annotation editor

bohandley committed:
fix annotation regression looking for tags before target

bohandley committed:
remove angular annotation partial

bohandley committed:
change ann tags type to string[] and use TagsInput to create ann

bohandley committed:
remove GraphiteEventsType, return annotations targets setting 'textEditor': true

bohandley committed:
fix yarn typecheck errors

chri2547 committed:
combines table viz docs (#52176)

nmarrs committed:
Canvas: Remove Open Editor option from context menu when in normal editor mode #52186

alexweav committed:
Alerting: Provisioning API respects global rule quota (#52180)

* Inject interface for quota service and create mock

* Check quota and return 403 if limit exceeded

* Implement tests for quota being exceeded

brendamuir committed:
Docs: fixes data source links (#52196)

* Docs: fixes data source links

* fixes unified alerting redirect

andresmgot committed:
AzureMonitor: Add ResourceGroups template variable (#52141)

aangelisc committed:
AzureMonitor: Improve handling of unsupported template variable cases in URIs (#52054)

* Set error message for certain template variable combinations

- Make use of setError method from query editor
- Update Azure Monitor error type
- Add test for case 2 from https://github.com/grafana/grafana/pull/51331

* Update template variable docs

* Fix lint issues

* Update docs/sources/datasources/azuremonitor/template-variables.md

Co-authored-by: Garrett Guillotte <100453168+gguillotte-grafana@users.noreply.github.com>

* PR comment updates

Co-authored-by: Garrett Guillotte <100453168+gguillotte-grafana@users.noreply.github.com>

svennergr committed:
Logs: Fixed LogContext being underneath the table (#52165)

ivanahuckova committed:
Loki: Add hint for pipeline error to query builder (#52134)

* WIP

* Update

* Add tests

* Fix lint/betterer errors

* Update public/app/plugins/datasource/loki/addToQuery.ts

* Update copy

eledobleefe committed:
Internationalisation: Translates the panel header menu (#52168)

aangelisc committed:
Cloud Datasources: Add necessary inputs and bump version for code coverage workflow (#52055)

* Add necessary inputs and bump version

* Test coverage commit

* Bump coverage workflow version

* Bump workflow version

* Try updated workflow

* Bump commit sha

* Rename empty file for testing

* Update sha

* Update sha

* Update sha

* Bump SHA

* Update SHA

* Update sha

* Bump SHA

* Bump SHA

* Bump sha

* Bump SHA

* Test all plugins (including backend)

* Bump SHA

* Update SHA

* Remove empty files and set appropriate workflow version

joshhunt committed:
Chore: Remove any that was introduced in a test (#52203)

grobinson-grafana committed:
Alerting: Fix bug where state did not change between Alerting and Error (#52204)

This commit fixes a bug where the state did not change from Alerting to Error if the evaluation result returned an error, or from Error to Alerting if evaluations stopped returning errors.

michal800106 committed:
Alerting: fix validation of alertmanager template. (#51538)

without setting function map from alertmanager we receive error:
method=PUT path=/api/v1/provisioning/templates/slack.message status=400
level=error msg="invalid object specification: invalid template: template: :1: function \"toUpper\" not defined"

So for validation we should use the same settings as alertmanager do
for templates internally.

kaydelaney committed:
Panels: Prevent panel content from escaping its bounds (#52167)

grafanabot committed:
ReleaseNotes: Updated changelog and release notes for 8.4.10

lpskdl committed:
fix: set RGL wrapper height to auto to remove empty space below the panel (#52122)

oscarkilhed committed:
SQL: Migrate (MS/My/Postgres)SQL configuration pages from Angular to React (#51891)

* Migrate SQL configuration pages from angular to react

* Move enums to types.ts and remove angular partials

* remove es lint disables and update betterer instead

* Fix automatically added type declarations

* Bump wor.. betterer ;)

* Export SecretInput component from grafana-ui

* Fix A11y issues

* Export SecretTextArea as well

* Fix typo

* Use const instead of var

* Fix typo in doc

* Add autoDetectFeatures to postgres config editor

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>
Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

sakjur committed:
Preferences: Get home dashboard from teams (#52225)

asimpson committed:
AzureMonitor: Update UI to experimental package (#52123)

* feat: make azure experimental the default

* feat: combine metrics query editor rows

fix: linter errors

* chore: remove test loop for DimensionFields test

kylebrandt committed:
SSE: Add noData type (#51973)

When there is a single frame with no fields (e.g. splunk datasource) SSE errors when trying to figure out the data type. This frame needs to exist since this is where the executedQueryString metadata exists.

This adds a new return type to SSE to represent no data, so the original frame with its metadata can still be maintained.

bohandley committed:
fix merge conflict

bohandley committed:
fix betterer

bohandley committed:
add dateTime for yarn fix to tests

IevaVasiljeva committed:
Access Control: Fix missing folder permissions (#52153)

* add the migration

* Update pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

Co-authored-by: Eric Leijonmarck <eric.leijonmarck@gmail.com>

bohandley committed:
fix incorrect merge conflict resolution

bohandley committed:
fix betterer
